Update Production Guide for Helm Chart docs (#23836)
mistercrunch/Airflow: docs/helm-chart/production-guide.rst

leveraging a cloud provider's database service such as AWS RDS is advised. Supported databases and versions
can be found at :doc:`Set up a Database Backend <apache-airflow:howto/set-up-database>`.

> Note: If you are using helm, you do not need to initialize the db 
> 
> .. code-block:: bash
> 
>     airflow db init
> 
> as outlined :doc:`Set up a Database Backend <apache-airflow:howto/set-up-database>`.

Reviewer: ```suggestion

.. note::

    When using the helm chart, you do not need to initialize the db with ``airflow db init``
    as outlined in :doc:`Set up a Database Backend <apache-airflow:howto/set-up-database>`.
```

Maybe like this instead?
